src/honey_badger.rs

```diff
@@ -567,23 +547,18 @@ where
     /// `epoch == Some(given_epoch)`.
     fn process_output(
         &mut self,
-        step: common_subset::Step<NodeUid>,
-        epoch: Option<u64>,
-    ) -> Result<FaultLog<NodeUid>> {
-        let common_subset::Step {
-            output,
-            mut fault_log,
-            mut messages,
-        } = step;
-        self.messages.extend_with_epoch(self.epoch, &mut messages);
-        // If this is the current epoch, the message could cause a new output.
-        if epoch.is_none() || epoch == Some(self.epoch) {
-            for cs_output in output {
-                fault_log.extend(self.send_decryption_shares(cs_output)?);
-                // TODO: May also check that there is no further output from Common Subset.
-            }
+        cs_step: common_subset::Step<NodeUid>,
+        epoch: u64,
+    ) -> Result<Step<C, NodeUid>> {
+        let mut step = Step::default();
+        let cs_outputs = step.extend_with(cs_step, |cs_msg| {
+            MessageContent::CommonSubset(cs_msg).with_epoch(epoch)
+        });
+        for cs_output in cs_outputs {
+            // There is at most one output.
```

Comment:
I'm fine to keep this as a comment but maybe the code could check that statement or ensure multiple outputs don't get `extend`ed so easily? What would be the best solution here? The function might return a `Fault` if there is a second output.

Reply:
I was thinking we might change `Step` and `DistAlgorithm` a bit, and essentially replace what's currently `Output` with either `Option<Output>` or `Vec<Output>`, depending on the algorithm. Not sure whether that's a good idea…

Reply:
I wouldn't return a fault: it can't be attributed to a particular node, and this could actually only happen due to a bug in the code. So it should be either an error or a panic, I guess.

Reply:
I made it an error log for now.